src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallerHelpers.cs

```diff
@@ -406,6 +406,19 @@ public static MarshalDirection GetMarshalDirection(TypePositionInfo info, StubCo
         }
 
         /// <summary>
+        /// Returns which stage cleanup should be performed for the parameter.
+        /// </summary>
+        public static StubCodeContext.Stage GetCleanupStage(TypePositionInfo info, StubCodeContext context)
+        {
+            if (context.Direction is MarshalDirection.UnmanagedToManaged)
+                return StubCodeContext.Stage.CleanupCallerAllocated;
+
+            if (GetMarshalDirection(info, context) is MarshalDirection.UnmanagedToManaged)
+                return StubCodeContext.Stage.CleanupCalleeAllocated;
+
+            return StubCodeContext.Stage.CleanupCallerAllocated;
```

Comment:
Can we rework this so we return the same value from one location. I think we should clarify why the fall through is the same as one of the special cases. This approach might be correct, but without comments here or additional details it looks rather odd.

Reply:
Yes, this is because we're not changing behavior for unmanaged to managed stubs, so it's a filter right away to use the previous behavior. I've added a comment and made the cases after that filter into a switch statement to make it clear.